packages/core/execution-context/core-execution-context-browser-internal/src/execution_context_service.ts

```diff
@@ -86,6 +88,13 @@ export class ExecutionContextService
       })
     );
 
+    // Track URL changes to make sure that we reflect the new path name
+    this.subscription.add(
+      history.listen((location) => {
+        start.set({ url: location.pathname });
```

Comment:
it might be that `location.pathname` didn't change, right? Maybe worth calling only when has changed?